evap/results/exporters.py

```diff
@@ -170,7 +173,10 @@ def filter_evaluations(
     def write_headings_and_evaluation_info(
         self,
         evaluations_with_results: list[tuple[Evaluation, OrderedDict[int, list[QuestionResult]]]],
-        semesters: Sequence[Semester] | QuerySet[Semester],
+        semesters: Sequence[Semester]
+        | QuerySet[
+            Semester
+        ],  # see https://github.com/typeddjango/django-stubs/issues/1924 why this is not a Collection
```

Comment:
1. I'd rather have your django issue linked here (https://code.djangoproject.com/ticket/35154), as the django-stubs issue is not about the core problem (they only document what django does/has).
2. I'd prefer to not have this comment (and those 4 "ugly" lines in general) copied around the codebase. I'd suggest we add a new generic helper type (`QuerySetLike` or `QuerySetOrContainer` or whatever) and use it whenever we want to express "QuerySet[x] or Container[x]". The comment would then go with this type definition.